avg_pool3d_backward: Port to structured

Freey0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59084 avg_pool3d_backward: Port to structured**

Differential Revision: [D28802619](https://our.internmc.facebook.com/intern/diff/D28802619)

Freey0 committed:
avg_pool3d_backward: Port to structured

[ghstack-poisoned]

Freey0 committed:
Update on "avg_pool3d_backward: Port to structured"




[ghstack-poisoned]

Freey0 committed:
Update on "avg_pool3d_backward: Port to structured"




[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28802619).

ezyang commented:
failure is real

Freey0 committed:
Update on "avg_pool3d_backward: Port to structured"


Differential Revision: [D28802619](https://our.internmc.facebook.com/intern/diff/D28802619)

[ghstack-poisoned]

Freey0 committed:
Update on "avg_pool3d_backward: Port to structured"


Differential Revision: [D28802619](https://our.internmc.facebook.com/intern/diff/D28802619)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28802619).

Freey0 committed:
Update on "avg_pool3d_backward: Port to structured"


Differential Revision: [D28802619](https://our.internmc.facebook.com/intern/diff/D28802619)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28802619).

ezyang commented:
more internal failures

```
 caffe2/aten/src/ATen/native/AveragePool3d.cpp:466:17: error: unused variable 'otime_for_shape_check' [-Werror,-Wunused-variable]
  const int64_t otime_for_shape_check = pooling_output_shape<int64_t>(itime, kT, padT, dT, 1, ceil_mode);
caffe2/aten/src/ATen/native/AveragePool3d.cpp:467:17: error: unused variable 'oheight_for_shape_check' [-Werror,-Wunused-variable]
  const int64_t oheight_for_shape_check = pooling_output_shape<int64_t>(iheight, kH, padH, dH, 1, ceil_mode);
                ^
caffe2/aten/src/ATen/native/AveragePool3d.cpp:468:17: error: unused variable 'owidth_for_shape_check' [-Werror,-Wunused-variable]
  const int64_t owidth_for_shape_check = pooling_output_shape<int64_t>(iwidth, kW, padW, dW, 1, ceil_mode);
```

Freey0 committed:
Update on "avg_pool3d_backward: Port to structured"


Differential Revision: [D28802619](https://our.internmc.facebook.com/intern/diff/D28802619)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28802619).

Freey0 committed:
Update on "avg_pool3d_backward: Port to structured"


Differential Revision: [D28802619](https://our.internmc.facebook.com/intern/diff/D28802619)

[ghstack-poisoned]
